### ISSUE TITLE
Checking within Eclipse (using the plugin) freezes Eclipse for about half a minute

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#302]
```
Hi,

I reported the problem on the mailing-list 
(https://groups.google.com/d/msg/checker-framework-discuss/p7rrYwPfONs/wjSSpfMDE
SgJ) but I think it is best to add it here :)

I am using the Checker Eclipse Plugin with Eclipse Kepler SR2.

I create a java project, add one class, empty.

I run the nullness checker and Eclipse freezes for some times until it gives 
back control to me.
I would expect it to be instantaneous.

If I add an error in the file, for example:

public class TestChecker {

    public void test2() {
        test().toString();
    }

    public String test() {
        return null;
    }

}

I notice that in the console the error is reported almost immediately as 
expected, but still, Eclipse stays frozen for about half a minute.

Something is happening but I don't know what: it makes the checker plugin 
unusable.

```

Original issue reported on code.google.com by `victor.n...@gmail.com` on 24 Mar 2014 at 10:38

**Comment by pwagland** [comment]
I am still seeing this issue, are there any thoughts as to what might be going on?

**Comment by gdhuse** [comment]
In my instance, I traced the pause to long task names when running the checker on a large group of files.  Here's my changeset, if you want to see if that fixes the issue for you: https://github.com/gdhuse/checker-framework/commit/e590989df54eac56e73c3ee64645fb2cc50e9ffc

**Comment by pwagland** [comment]
Thanks for the update @gdhuse . Sadly I don't think that this is my problem, since I am getting the issue that the entire UI locks up, and all 8 cores on my machine are pegging to 100% usage. It does this for 1/2 an hour or more. I get to see the message that you reduce in size fairly quickly, it is after this message shows that the UI really locks up.

**Comment by wmdietl** [comment]
@gdhuse Please do submit a pull request with your change!
@pwagland Can you reproduce the problem on the command line? Then we know whether it's a problem with the Eclipse plug-in or the Checker Framework.

**Comment by mernst** [comment]
Since there have been no replies to the request for further information, I am closing this pull request.
